### DISCUSSION TITLE
How can I create Jasper Ai using Python?

### DISCUSSION THREAD

**Discussion by hasnatkhan786** [sherlock-project/sherlock#1641] [General]
Hey experts[,](https://balloonslane.com/nyc-balloon/) 
I am trying to create a Jasper Ai bot. I tried different courses but was unable to understand them. Let me know of any easy method and easy course.

**Comment by jayyycol** [comment]
name: Pull Request Action

on:
  pull_request:
    branches: [ master ]

jobs:
  getchange:
    runs-on: ubuntu-latest
    outputs:
      matrix: ${{ steps.changes.outputs.matrix }}
    steps:
      - id: changes
        run: |
          URL="https://api.github.com/repos/sherlock-project/sherlock/pulls/${{ github.event.pull_request.number }}/files"
          FILES=$(curl -s -X GET -G $URL | jq -r '.[] | .filename')
          if echo $FILES | grep -q ".json"; then
            echo "::set-output name=matrix::{\"include\":[{\"python\":\"3.8\"}]}"
          else
            echo "::set-output name=matrix::{\"include\":[{\"python\":\"3.7\"},{\"python\":\"3.8\"}]}"
          fi
  build:
    needs: [getchange]
    runs-on: ubuntu-latest
    strategy:
      matrix: ${{ fromJson(needs.getchange.outputs.matrix) }}

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python ${{ matrix.python }}
      uses: actions/setup-python@v1
      with:
        python-version: ${{ matrix.python }}
    - name: Install Dependencies
      run: |
        python -m pip install --upgrade pip
        pip install flake8 pytest
        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
    - name: Lint With flake8
      run: |
        # stop the build if there are Python syntax errors or undefined names
        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics

        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
    - name: Sherlock Site Detect Tests
      run: |
        cd sherlock && python -m unittest tests.all.SherlockDetectTests --verbose
